### ISSUE TITLE
MGLPointAnnotation Hashable conformance

### ISSUE THREAD

**Issue by ollie-eman** [mapbox/mapbox-gl-native#6677]
**Platform:** iOS
**Mapbox SDK version:** 3.3.4
### Steps to trigger behavior

I've created a subclass of `MGLPointAnnotation` named `MyCustomAnnotation`

``` swift
class MyCustomAnnototation: MGLPointAnnotation { }
```

and defined a `Set` to store the annotations, to ensure I don't add any duplicates.

``` swift
var annotations = Set<MyCustomAnnotation>()
```

Next i've created two annotations:

``` swift
let coordinate = CLLocationCoordinate2D(latitude: 51.50, longitude: -0.12)
let annotationOne = MyCustomAnnotation(location: coordinate)
let annotationTwo = MyCustomAnnotation(location: coordinate)
```

To insert the annotations:

``` swift
var insertionResult = annotations.insert(annotationOne)
if insertionResult.inserted {
    print("Annotation one inserted")
}

insertionResult = annotations.insert(annotationTwo)
if insertionResult.inserted {
    print("Annotation two inserted")
}
```
### Expected behavior

The first annotation (`annotationOne`) should be inserted successfully into the Set. The second annotation (`annotationTwo`) should not be inserted as an annotation with the same coordinates already exists.
### Actual behavior

Both annotations are added to the `Set`

**Comment by boundsj** [comment]
Thanks @ollie-eman. I think this will be fixed when https://github.com/mapbox/mapbox-gl-native/pull/6559 lands. Although, note that in that implementation the hash would not be just a function of coordinate latitude and longitude as you've noted above.

**Comment by ollie-eman** [comment]
Great thanks @boundsj. Taking into consideration the hash implementation, do you think it will fix the issue i'm having and enable me to store the annotations in a Set? Also, do you know when it might be available via Cocoapods?

**Comment by boundsj** [comment]
> do you think it will fix the issue i'm having and enable me to store the annotations in a Set

From what I can tell, yes. Assuming `MyCustomAnnotation` is the "same" in the context of `MGLPointAnnotation`'s `hash` or your subclass's override.

> do you know when it might be available via Cocoapods

#6559 fixes #6200 which is currently [planned for our next release](https://github.com/mapbox/mapbox-gl-native/milestone/37).

**Comment by 1ec5** [comment]
Fixed in #6559.
